### PR TITLE
Fix platform tool dependency determination

### PR DESCRIPTION
**Pull request by per1234**
**Please check if the PR fulfills these requirements**

- [x] The PR has no duplicates (please search among the [Pull Requests](https://github.com/arduino/arduino-cli/pulls)
      before creating one)
- [x] The PR follows
      [our contributing guidelines](https://arduino.github.io/arduino-cli/latest/CONTRIBUTING/#pull-requests)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?**

Bug fix.
- **What is the current behavior?**

The platform tool dependency check was giving false negatives due to comparing pointers instead of version values.

This caused tools to be removed during platform uninstallation even when another installed platform had a dependency on
that tool.
* **What is the new behavior?**

The `semver` package's `Equal` method is used for the comparison, resulting in accurate tool dependency determination.
- **Does this PR introduce a breaking change?**

No
* **Other information**:

Steps to reproduce the bug fixed by this PR:
```
arduino-cli core install arduino-beta:mbed
arduino-cli core install arduino:samd
arduino-cli core uninstall arduino:samd
arduino-cli sketch new /tmp/Foo
arduino-cli compile -b arduino-beta-development:mbed:envie_m7 /tmp/Foo
```
`arduino-cli core uninstall arduino:samd` removes `arduino:arm-none-eabi-gcc@7-2017q4`, even though it's a dependency of the installed `arduino-beta:mbed` platform, causing the compilation for any board of that platform to fail.

### CODE REVIEW

**Review by cmaglie — APPROVED** [review]

**Review by silvanocerza — COMMENTED** [review]
Great job as always Per!
Would be awesome if you could add an integration test too, I feel it's an important thing we should test for.

**Comment by per1234** [comment]
>Would be awesome if you could add an integration test too

Excellent suggestion @silvanocerza! I have now added an integration test: https://github.com/arduino/arduino-cli/pull/1020/commits/b13a29c1ad94838ed037d59d324bc4098af1973c

File: test/test_core.py [hunk 1]
```diff
@@ -205,6 +205,49 @@ def test_core_uninstall(run_command):
...
+
+    result = run_command("config dump --format json")
+    assert result.ok
+    settings_json = json.loads(result.stdout)
+    arduino_tools_path = os.path.join(
+        settings_json["directories"]["data"],
+        "packages",
+        "arduino",
+        "tools"
+    )
```
Comment (silvanocerza) [inline test/test_core.py:224]: There's no need to load the settings like so, you already have access to these paths through the `data_dir`, `working_dir` and `downloads_dir` fixtures, you can find them on the `tests/conftest.py` file, they're also used for some tests in this file.

I also suggest `pathlib.Path` to join paths and check those files exist.

**Review by silvanocerza — APPROVED** [review]

File: test/test_core.py (hunk 1 above)
Comment (per1234) [reply test/test_core.py]: Thanks for your advice Silvano. I see you already noticed I updated the PR accordingly, so I'll mark this "Resolved".

**Pull request merged**